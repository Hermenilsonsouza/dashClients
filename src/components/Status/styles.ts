import styled, { css } from 'styled-components'
import { StatusProps } from '.'

export const Wrapper = styled.main`
  padding: 16px;
  display: flex;
  gap: 8px;
  align-items: center;
`

export const Title = styled.div`
  font-size: 18px;
  color: #333;
`

const WrapperModifies = {

  Ativo: () => css`
    background-color: green;
  `,

  Inativo: () => css`
  background-color: red;
  `,

  Desativado: () => css`
  background-color: gray;
  `,

'Aguardando ativação': () => css`
  background-color: #EEAD2D;
  `,
}

export const StatusBall = styled.div<StatusProps>`
  width: 16px;
  height: 16px;
  border-radius: 8px;

  ${({status}) => css`

    ${!!status && WrapperModifies[status]}

  `}
`

