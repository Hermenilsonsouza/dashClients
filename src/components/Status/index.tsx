import * as S from './styles'

export type StatusProps = {
  status: 'Ativo' | 'Inativo' | 'Aguardando ativação' | 'Desativado'
}

const Status = ({ status }: StatusProps) => (
  <S.Wrapper>
    <S.StatusBall status={status} />
    <S.Title>{status}</S.Title>
  </S.Wrapper>
)

export default Status
