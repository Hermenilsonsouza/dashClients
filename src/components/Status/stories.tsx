import { Meta, StoryObj } from '@storybook/react'
import Status from '.'

export default {
  title: 'Status',
  component: Status
} as Meta

export const Default: StoryObj = {}
