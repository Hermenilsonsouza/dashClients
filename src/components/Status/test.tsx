import { render, screen } from '@testing-library/react'

import Status from '.'

describe('<Status />', () => {
  it('should render the Status', () => {
    const { container } = render(<Status status="Aguardando ativação" />)

    expect(screen.getByText(/Aguardando ativação/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
