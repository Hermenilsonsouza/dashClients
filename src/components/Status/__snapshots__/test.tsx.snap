// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> should render the Status 1`] = `
.c0 {
  padding: 16px;
  display: flex;
  gap: 8px;
  align-items: center;
}

.c2 {
  font-size: 18px;
  color: #333;
}

.c1 {
  width: 16px;
  height: 16px;
  border-radius: 8px;
  background-color: #EEAD2D;
}

<main
  class="c0"
>
  <div
    class="c1"
    status="Aguardando ativação"
  />
  <div
    class="c2"
  >
    Aguardando ativação
  </div>
</main>
`;
